app/views.py

```diff
@@ -201,3 +274,35 @@ def default(self, thing):
 
 def jsonize(x):
     return json.dumps(x, cls=DBModelJSONEncoder)
+
+# ----------------------------------------------------------- utility fns --- #
+
+def mkdirp(p):
+    if os.path.exists(p) and os.path.isdir(p):
+        return False
+    else:
+        os.makedirs(p)
+        return True
+
+
+def upload_dir(p):
+    if isinstance(p, basestring):
+        add_path = (p,) # Turn the single element into a tuple that can be
+                        # joined into a full path
+    elif isinstance(p, (list, tuple)):
```

Comment:
Good catch!

Sent from my iPhone

> On Aug 21, 2016, at 2:06 PM, Marc Daya <notifications@github.com> wrote:
> 
> In app/views.py:
> 
> > +
> > +# ----------------------------------------------------------- utility fns --- #
> > +
> > +def mkdirp(p):
> > +    if os.path.exists(p) and os.path.isdir(p):
> > +        return False
> > +    else:
> > +        os.makedirs(p)
> > +        return True
> > +
> > +
> > +def upload_dir(p):
> > +    if isinstance(p, basestring):
> > +        add_path = (p,) # Turn the single element into a tuple that can be
> > +                        # joined into a full path
> > +    elif isinstance(p, (list, tuple)):
> Perhaps too generic. Dictionaries and and other structures are iterable, but can't sensibly be turned into a path.
> 
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>